### PR TITLE
DSM: Add disabled dropdown item state

### PR DESCRIPTION
**Pull request by ValentinMumble**
**Description (for Contributor and Core Developer)**

image
image
image

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

**Review by SamirBoulil — APPROVED** [review]
Should we test we can't click on a disabled item ? or check that it's just disabled ?

File: akeneo-design-system/src/components/Dropdown/Item/Item.tsx [hunk 1]
```diff
@@ -119,9 +139,10 @@ const Item = React.forwardRef<HTMLDivElement, ItemProps>(
     return (
       <ItemContainer
         tall={tall}
-        tabIndex={actionableRef.current === null ? 0 : -1}
+        tabIndex={null === actionableRef.current ? 0 : -1}
         onClick={handleClick}
         onKeyDown={handleKeyDown}
+        disabled={disabled}
```
Comment (StevenVAIDIE) [inline akeneo-design-system/src/components/Dropdown/Item/Item.tsx:145]:
```suggestion
        disabled={disabled}
        aria-disabled={disabled}
```

File: akeneo-design-system/src/components/Dropdown/Item/Item.tsx [hunk 2]
```diff
@@ -119,9 +139,10 @@ const Item = React.forwardRef<HTMLDivElement, ItemProps>(
     return (
       <ItemContainer
         tall={tall}
-        tabIndex={actionableRef.current === null ? 0 : -1}
+        tabIndex={null === actionableRef.current ? 0 : -1}
```
Comment (StevenVAIDIE) [inline akeneo-design-system/src/components/Dropdown/Item/Item.tsx:142]: To disable the keyboard navigation

```suggestion
        tabIndex={null === actionableRef.current && !disabled ? 0 : -1}
```

File: akeneo-design-system/src/components/Dropdown/Item/Item.tsx [hunk 3]
```diff
@@ -1,55 +1,62 @@
...
-  &:focus ${ItemLabel} {
-    color: ${getColor('grey', 120)};
-  }
-  &:hover {
-    background: ${getColor('grey', 20)};
-  }
-  &:hover ${ItemLabel} {
-    color: ${getColor('brand', 140)};
-  }
-  &:active ${ItemLabel} {
```
Comment (StevenVAIDIE) [inline akeneo-design-system/src/components/Dropdown/Item/Item.tsx:40]: Problem now is that badge content is also in italic and bold

**Review by StevenVAIDIE — APPROVED** [review]

**Comment by StevenVAIDIE** [comment]
Visual test does not detect error on disabled items :cry:

File: akeneo-design-system/src/components/Dropdown/Item/Item.tsx (hunk 2 above)
Comment (ValentinMumble) [reply akeneo-design-system/src/components/Dropdown/Item/Item.tsx:142]: I tried at first but it make the following Item unreachable with the arrow navigation, so I decided to leave it navigable but not actionable

**Review by juliensnz — APPROVED** [review]

**Pull request merged**